### PR TITLE
Implementation of case function.

### PR DESCRIPTION
**Pull request by lukasz-soszynski-eliatra**
### Description
Implementation of the `case` function

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/645

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLEvalITSuite.scala [hunk 1]
```diff
@@ -504,7 +506,63 @@ class FlintSparkPPLEvalITSuite
     comparePlans(logicalPlan, expectedPlan, checkAnalysis = false)
   }
 
+  test("eval case function") {
```
Comment (YANG-DB) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLEvalITSuite.scala:509]: Can you also add more complex tests use case including composition of additional commands such as where conditions and stats ?
thanks

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLFiltersITSuite.scala [hunk 2]
```diff
@@ -348,4 +348,38 @@ class FlintSparkPPLFiltersITSuite
     assert(compareByString(expectedPlan) === compareByString(logicalPlan))
   }
 
+  test("case function used as filter") {
```
Comment (YANG-DB) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLFiltersITSuite.scala:351]: can u also add here some additional tests use cases of more complicated composition ppl commands ?

File: ppl-spark-integration/README.md [hunk 3]
```diff
@@ -265,6 +266,15 @@ Assumptions: `a`, `b`, `c` are existing fields in `table`
  - `source = table | eval f = ispresent(a)`
  - `source = table | eval r = coalesce(a, b, c) | fields r`
  - `source = table | eval e = isempty(a) | fields e`
+   ```
```
Comment (YANG-DB) [inline ppl-spark-integration/README.md:269]: plz add more examples for using case in other use cases that include where clause / stats commands

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ast/AbstractNodeVisitor.java [hunk 4]
```diff
@@ -171,6 +171,8 @@ public T visitIsEmpty(IsEmpty node, C context) {
     return visitChildren(node, context);
   }
 
+  // TODO add case
```
Comment (YANG-DB) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ast/AbstractNodeVisitor.java:174]: why is it TODO ?

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ast/AbstractNodeVisitor.java (hunk 4 above)
Comment (lukasz-soszynski-eliatra) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ast/AbstractNodeVisitor.java:174]: Good catch. I need to remove this TODO.

**Comment by YANG-DB** [comment]
@lukasz-soszynski-eliatra  please ping me when this is ready for review
thanks

**Comment by lukasz-soszynski-eliatra** [comment]
I'll do my best to make corrections today.

**Comment by lukasz-soszynski-eliatra** [comment]
@YANG-DB 
Please take a look at the [commit](https://github.com/opensearch-project/opensearch-spark/pull/695/commits/5dc26872d4f678e899d978648e99f491bde56828). I added more tests and documentation.

**Review by YANG-DB — APPROVED** [review]

**Comment by LantaoJin** [comment]
@YANG-DB does this syntax match what we expected? From the issue description, the valid synatx is:
```
CASE(<predicate1>, <value1>, [<predicate2>, <value2>, ...] [ELSE] <default>)
```
An valid example:
```
source = my_index
| eval status_category =
    case(status_code >= 200 AND status_code < 300, 'Success',
         status_code >= 300 AND status_code < 400, 'Redirection'
         status_code >= 400 AND status_code < 500, 'Client Error'
         status_code >= 500, 'Server Error',
         'Unknown')
| stats count() by status_category
```

But the implementation we merged is totally different as:
```
CASE(<predicate1>, <value1>[, <predicate2>, <value2>, ...] [ELSE <default>])
```
The example above needs rewritten to
```
source = my_index
| eval status_category =
    case(status_code >= 200 AND status_code < 300, 'Success',
         status_code >= 300 AND status_code < 400, 'Redirection'
         status_code >= 400 AND status_code < 500, 'Client Error'
         status_code >= 500, 'Server Error'
         ELSE 'Unknown')
| stats count() by status_category
```
Note, the `,` after `'Server Error'` is removed and `ELSE` keyword is required for this case.
Please double confirm the current syntax is by designed.

**Pull request merged**